common/utils.c

```diff
@@ -173,3 +175,35 @@ void *tal_dup_talarr_(const tal_t *ctx, const tal_t *src TAKES, const char *labe
 	}
 	return tal_dup_(ctx, src, 1, tal_bytelen(src), 0, label);
 }
+
+/* Check for valid UTF-8 */
+bool utf8_check(const void *vbuf, size_t buflen)
+{
+	const u8 *buf = vbuf;
+	struct utf8_state utf8_state = UTF8_STATE_INIT;
+	bool need_more = false;
+
+	for (size_t i = 0; i < buflen; i++) {
+		if (!utf8_decode(&utf8_state, buf[i])) {
+			need_more = true;
```

Comment:
is there no early exit for invalid utf-8? the `need_more` flag flipping is a bit unintuitive

Reply:
utf8_decode is a bit weird:

```C
/*  
 * Returns false if it needs another character to give results.
 * Otherwise returns true, @utf8_state can be reused without initializeation,
 * and sets errno:
 * 0: success
 * EINVAL: bad encoding.
 * EFBIG: not a minimal encoding.
 * ERANGE: encoding of invalid character.
*/
```